features/filesystem/storagelite/source/StorageLiteFS.cpp

```diff
@@ -0,0 +1,320 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "StorageLiteFS.h"
+#include "mbed_retarget.h"
+#include "string.h"
+#include "stdlib.h"
+
+typedef struct {
+    char *file_name;
+    int open_flags;
+} file_handle_data_t;
+
+// Convert basic StorageLite error numbers to errno based ones
+static int convert_to_errno(int err)
+{
+    switch (err) {
+        case STORAGELITE_SUCCESS:
+            return  0;
+        case STORAGELITE_READ_ERROR:
+            return -EIO;
+        case STORAGELITE_WRITE_ERROR:
+            return -EIO;
+        case STORAGELITE_NOT_FOUND:
+            return -ENOENT;
+        case STORAGELITE_DATA_CORRUPT:
+            return -EILSEQ;
+        case STORAGELITE_BAD_VALUE:
+            return -EINVAL;
+        case STORAGELITE_BUFF_TOO_SMALL:
+            return -ENOMEM;
+        case STORAGELITE_NO_SPACE_ON_BD:
+            return -ENOSPC;
+        case STORAGELITE_OS_ERROR:
+            return -EINTR; // FIXME: Find a better error
+        case STORAGELITE_UNINITIALIZED:
+            return -ENXIO; // FIXME: Find a better error
+        case STORAGELITE_MAX_FILES_REACHED:
+            return -ENFILE;
+        case STORAGELITE_NOT_SUPPORTED:
+            return -ENOSYS;
+        default:
+            return -err;
+    }
+}
+
+StorageLiteFS::StorageLiteFS(const char *name, StorageLite *stlite, uint32_t flags)
+    : FileSystem(name), _stlite(stlite), _flags(flags), _bd(0)
+{
+    MBED_ASSERT(stlite);
+    mount(_bd);
+}
+
+StorageLiteFS::~StorageLiteFS()
+{
+    unmount();
+}
+
+int StorageLiteFS::mount(BlockDevice *bd)
+{
+    _bd = bd;
+    int ret = _stlite->init(bd);
+    return convert_to_errno(ret);
+}
+
+int StorageLiteFS::unmount()
+{
+    int ret = _stlite->deinit();
+    _bd = 0;
+    return convert_to_errno(ret);
+}
+
+int StorageLiteFS::reformat(BlockDevice *bd)
+{
+    if (bd) {
+        _bd = bd;
+    }
+
+    int ret = _stlite->init(_bd);
+    if (ret) {
+        return convert_to_errno(ret);
+    }
+
+    ret = _stlite->reset();
+    if (ret) {
+        return convert_to_errno(ret);
+    }
+
+    return 0;
+}
+
+int StorageLiteFS::remove(const char *filename)
+{
+    int ret = _stlite->remove((const uint8_t *) filename, strlen(filename) + 1);
```

Comment:
Error handling/checking- filename

Reply:
Disagree, but the casts suggest a type problem - you should be passing filenames around as `char *` pointers, just like any string. If you need to cast to `uint8_t *` it should be lower level than this.

It this is converting to an opaque key, the opaque key's type should probably be `void *` to make caller's lives easier. And I wouldn't have thought you needed to include the NUL byte as part of the key, if you're giving explicit key lengths.

Reply:
Right. The idea was to have opaque keys. will change to `void *` and remove the null terminator.